honeybadgermpc/hbavss.py

```diff
@@ -229,57 +232,144 @@ def __exit__(self, typ, value, traceback):
         for task in self.tasks:
             task.cancel()
 
-    async def _process_avss_msg(self, avss_id, dealer_id, dispersal_msg):
+    async def _process_avss_msg(self, avss_id, dealer_id, rbc_msg, avid):
         tag = f"{dealer_id}-{avss_id}-B-AVSS"
         send, recv = self.get_send(tag), self.subscribe_recv(tag)
 
         def multicast(msg):
             for i in range(self.n):
                 send(i, msg)
 
-        commitments, ephemeral_public_key, encrypted_witnesses = loads(
-            dispersal_msg)
+        # get phi and public key from reliable broadcast msg
+        commitments, ephemeral_public_key = loads(rbc_msg)
+        # retrieve the z
+        dispersal_msg = await avid.retrieve(tag, self.my_id)
+        encrypted_witnesses = loads(dispersal_msg)
+
         secret_size = len(commitments)
 
         # all_encrypted_witnesses: n
         shared_key = pow(ephemeral_public_key, self.private_key)
 
         shares = [None] * secret_size
+        auxes = [None] * secret_size
+        witnesses = [None] * secret_size
         # Decrypt
         for k in range(secret_size):
-            shares[k], cur_aux, witness = SymmetricCrypto.decrypt(
+            shares[k], auxes[k], witnesses[k] = SymmetricCrypto.decrypt(
                 str(shared_key).encode("utf-8"), encrypted_witnesses[k])
             if not self.poly_commit.verify_eval(
-                    commitments[k], self.my_id+1, shares[k], cur_aux, witness):
-                # will be replaced by sending out IMPLICATE message later
-                # multicast(HbAVSSMessageType.IMPLICATE)
-                logging.error("PolyCommit verification failed.")
-                raise HoneyBadgerMPCError("PolyCommit verification failed.")
+                    commitments[k], self.my_id+1, shares[k], auxes[k], witnesses[k]):
+                # invalid share found and send all implicate msg
+                multicast((HbAVSSMessageType.IMPLICATE, self.private_key, k))
 
-        multicast(HbAVSSMessageType.OK)
+        multicast((HbAVSSMessageType.OK, ""))
 
-        # Bracha-style agreement
-        # to handle the case that a (malicious) node sends
-        # an OK or READY message multiple times
         ok_set = set()
         ready_set = set()
-        # if 2t+1 OK or t+1 READY -> send all ready
-        while len(ok_set) < 2 * self.t + 1 and len(ready_set) < self.t + 1:
-            sender, avss_msg = await recv()  # First value is the `sid`
-            if avss_msg == HbAVSSMessageType.OK and sender not in ok_set:
-                ok_set.add(sender)
-            elif avss_msg == HbAVSSMessageType.READY and sender not in ready_set:
-                ready_set.add(sender)
-
-        multicast(HbAVSSMessageType.READY)
+        r1_set = set()
```

Comment:
what's `r1_set`? I am not aware of the protocol but can you pick a more descriptive name?

Reply:
That message is recovery phase 1 and it is called exactly as R1 in the paper